keps/sig-node/5394-psi-node-conditions/README.md

```diff
@@ -0,0 +1,587 @@
+Integration tests allow control of the configuration parameters used to start the binaries under test.
+This is different from e2e tests which do not allow configuration of parameters.
+Doing this allows testing non-default options and multiple different and potentially conflicting command line options.
+-->
+
+<!--
+This question should be filled when targeting a release.
+For Alpha, describe what tests will be added to ensure proper quality of the enhancement.
+
+For Beta and GA, add links to added tests together with links to k8s-triage for those tests:
+https://storage.googleapis.com/k8s-triage/index.html
+-->
+
+Any identified external user of either of these endpoints (prometheus, metrics-server) should be tested to make sure they're not broken by new fields in the API response. 
+
+##### e2e tests
+
+<!--
+This question should be filled when targeting a release.
+For Alpha, describe what tests will be added to ensure proper quality of the enhancement.
+
+For Beta and GA, add links to added tests together with links to k8s-triage for those tests:
+https://storage.googleapis.com/k8s-triage/index.html
+
+We expect no non-infra related flakes in the last month as a GA graduation criteria.
+-->
+
+Test plan:
+- Enable the feature gate.
+- For each of memory and IO
+  - Schedule workloads to a node that overwhelms the resource.
+  - Ensure the node is marked with corresponding PSI pressure node condition and taint.
+  - Create more workloads and observe that the workloads cannot be scheduled onto this node.
+  - Delete existing workloads to free up the resource.
+  - Ensure the node condition and taint are removed.
+  - Create more workloads and observe the the workloads can be scheduled.
+  - Clean up.
+
+### Graduation Criteria
+
+#### Alpha
+
+- Conduct experiments to decide the scope of the node condition (system v.s. kubepods v.s. node) as well as the default threshold
+- Enables kubelet to 
+report node conditions based off PSI values.
+- Initial e2e tests completed and enabled if CRI implementation supports
+it.
+- Add documentation for the feature.
+
+#### Beta
+
+- Decide whether CPU PSI will be used for node conditions.
+- Feature gate is enabled by default.
+- Extend e2e test coverage.
+- Allowing time for feedback.
+
+#### GA
+- TBD
+
+#### Deprecation
+
+- Announce deprecation and support policy of the existing flag
+- Two versions passed since introducing the functionality that deprecates the flag (to address version skew)
+- Address feedback on usage/changed behavior, provided on GitHub issues
+- Deprecate the flag
+-->
+
+### Upgrade / Downgrade Strategy
+
+No impact. Runc will be upgraded to 1.2.0 version as a prerequisite for this feature,
+and all the other components will already be at expected levels. Hence there shouldn't
+be a problem in upgrading or downgrading. Besides, it's always possible to upgrade/downgrade
+to a different kubelet version.
+
+### Version Skew Strategy
+
+N/A
```

Comment:
Since you're changing several components, I'd expect a description how various versions of these components will affect the functionality of this feature.

Reply:
Added version skew details. PTAL